### DISCUSSION TITLE
Add support for third party Map field types with ICS export

### DISCUSSION THREAD

**Discussion by ademers** [solspace/craft-calendar#80] [Feature Requests]
**Describe the bug or issue you're experiencing**

ICS Export Error: strip_tags() expects parameter 1 to be string, object given Related to ICS Location Field

When I attempt to export events as an ICS file, I get the following error:

```
strip_tags() expects parameter 1 to be string, object given

 in /home/vagrant/sites/lccphoto/vendor/solspace/craft3-calendar/packages/plugin/src/Library/Export/ExportCalendarToIcs.php at line 92
8384858687888990919293949596979899100101

 
        $exportString .= "BEGIN:VEVENT\r\n";
        $exportString .= sprintf("UID:%s@solspace.com\r\n", $uidHash);
        $exportString .= sprintf("DTSTAMP:%s\r\n", $this->now->format(self::DATE_TIME_FORMAT));
 
        if ($description) {
            $exportString .= sprintf("DESCRIPTION:%s\r\n", $this->prepareString(strip_tags($description)));
        }
        if ($location) {
            $exportString .= sprintf("LOCATION:%s\r\n", $this->prepareString(strip_tags($location)));
        }
 
        if ($event->isAllDay()) {
            $exportString .= sprintf("DTSTART;VALUE=DATE:%s\r\n", $startDate->format(self::DATE_FORMAT));
            $exportString .= sprintf(
                "DTEND;VALUE=DATE:%s\r\n",
                $endDate->copy()->addDay()->format(self::DATE_FORMAT)
            );
        } elseif ('UTC' === $timezone) {

```
I'm using the [Maps plugin](https://github.com/ethercreative/simplemap) which provides a field type that I'm using for event locations. It seems that the plugin is generating and object instead of a string expected by Calendar

**Steps to reproduce**

1. Install Maps plugin and create a field called **Event Location** with Map field type.
2. Assign **Event Location** field to ICS Location field in one or more Calendars.
3. Create ICS Export template.
4. Attempt to download ICS file.
5. See error above.

**Expected behavior**

Download ICS file containing event location without errors.

**Craft & Plugin Info (please complete the following information):**

- Craft Version: Craft Pro 3.7.11
- Calendar Version: 3.3.5
- Calendar Edition: Pro
- Fresh Install or Upgrade:

**Additional context**

Maps plugin 3.9.2 Lite edition

**Comment by kjmartens** [comment]
Sorry for the delay @ademers. Unfortunately you currently cannot use a field like Smart Maps for the ICS export location setting. Calendar will only work with regular text/textarea type fields at this time. We do hope to add support for complex field types in the future though. 🙂

**Comment by ademers** [comment]
@kjmartens Thanks! So, as a possible workaround, I could create a new text field called **Event Location ICS** and leave it blank/empty? Or, does it need to be populated?

**Reply by kjmartens** [reply]
You could just choose not to select a field for this setting. 🙂

**Reply by ademers** [reply]
I see the **None** option now. Thanks!

**Comment by ademers** [comment]
During the recent dotAll, P&T announced a new Address Field coming in Craft 4. Are you planning on adding support for it? I assume it could be used as the Event Location for ICS exports. https://craftquest.io/dotall#state-of-craft-commerce

**Reply by kjmartens** [reply]
It's very likely we will add support for this, especially being an official field type, in Calendar for Craft 4. I can't guarantee it though, because we haven't see what it looks like, but I'm sure there won't be an issue. 🙂
